warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RootNegationCheckVisitor.java

```diff
@@ -46,8 +46,7 @@ private boolean hasLeafWithoutNegation(JexlNode node) {
             
             if (ASTNENode.class.equals(childClass) || ASTNotNode.class.equals(childClass) || ASTNRNode.class.equals(childClass)) {
                 // no need to inspect this path anymore
-                continue;
-            } else if (child.jjtGetNumChildren() == 0) {
+            } else if (child == null || child.jjtGetNumChildren() == 0) {
```

Comment:
A null counts as a leaf node?  Is that really true?  Should we not just throw an exception if we encounter a null child?  This is a functionality change.

Reply:
I'm fine with letting this blow up on a null child. Will revert this line back